Mlem/Views/Tabs/Posts/Components/Post/Components/Instance and Community List/Add Account View.swift

```diff
@@ -235,13 +235,13 @@ struct AddSavedInstanceView: View
                 switch error {
                 case EndpointDiscoveryError.couldNotFindAnyCorrectEndpoints:
                     message = "Could not connect to \(instanceLink)"
-                case APIRequestError.encoding:
-                    message = "Invalid credentials"
                 case UserIDRetrievalError.couldNotFetchUserInformation:
                     message = "Mlem couldn't fetch you account's information.\nFile a bug report."
+                case APIClientError.encoding:
+                    message = "Invalid credentials"
```

Comment:
Why do we return this on an encoding error? Why don't we look at the 403 HTTP status code?

Reply:
Ah, this is perhaps just bad wording on my part, an`.encoding` error means we failed to parse to body of the request _before_ sending... we should raise in the creative channel that it'd be worth developing a _tone_ for messaging in the app so we can all agree on copy for different error scenarios... any thoughts on this one?

Something like:

> Please check your username and password

Reply:
I think that's a better message, yeah.

And if this is before sending, why are we checking it here? Why are we not checking the form elsewhere?

I think this should at least get a TODO to abstract out more into the UX side of things, so we don't get partial values.

Reply:
yeah, the inputs need validatio/sanitised as currently they're not - but I agree that's a job for another PR, I'll add a TODO 👍

Reply:
d37f4ea4e109ff48bab2c41ecee5c57c9a3d76a8